spec/javascripts/models/beta/story_spec.js

```diff
@@ -246,107 +246,179 @@ describe('Story model', function () {
+              const newAttributes = { storyType: otherNotFeatureType }
 
-      expect(changedStory).toEqual({
-        _editing: {
-          labels: [],
-          storyType: 'feature',
-          estimate: newAttributes.estimate,
-          state: 'unstarted',
-          _isDirty: true
-        }
-      });
-    });
+              const changedStory = Story.editStory(story, newAttributes);
 
-    const estimateValues = [1, 2, 3, 5, 8];
+              expect(changedStory._editing.storyType).toEqual(otherNotFeatureType);
+              expect(changedStory._editing.estimate).toEqual('');
+            });
+          });
+        });
 
-    estimateValues.forEach(estimate => {
-      it(`change story state to unstarted when estimate is ${estimate} and sets _isDirty to true`, () => {
-        const story = { _editing: { storyType: 'feature', estimate: '', state: 'unscheduled' } };
-        const newAttributes = { estimate, };
+        Object.keys(status).forEach(item => {
+          describe(`when new status is ${status[item]}`, () => {
+            it(`change status to ${status[item]}`, () => {
+              const story = { _editing: { storyType: noFeatureType } };
+              const newAttributes = { state: status[item] }
 
-        const changedStory = Story.editStory(story, newAttributes);
+              const changedStory = Story.editStory(story, newAttributes);
 
-        expect(changedStory).toEqual({
-          _editing: {
-            labels: [],
-            storyType: 'feature',
-            estimate,
-            state: 'unstarted',
-            _isDirty: true
-          }
+              expect(changedStory._editing.state).toEqual(status[item]);
+            });
+          });
         });
-      });
 
-      it(`change story state to unscheduled when estimate is null and sets _isDirty to true`, () => {
-        const story = { _editing: { storyType: 'feature', estimate, state: 'unstarted' } };
-        const newAttributes = { estimate: '' };
+        describe(`when story status is ${UNSCHEDULED}`, () => {
+          describe(`when new story type is feature`, () => {
+            it('change estimate to "" and storyType to feature', () => {
+              const story = { _editing: { 
+                storyType: noFeatureType, 
+                estimate: '', 
+                state: UNSCHEDULED } 
+              };
+              const newAttributes = { storyType: FEATURE }
+    
+              const changedStory = Story.editStory(story, newAttributes);
+    
+              expect(changedStory._editing.storyType).toEqual(FEATURE);
+              expect(changedStory._editing.state).toEqual(UNSCHEDULED);
+              expect(changedStory._editing.estimate).toEqual('');
+            });
+          });
+        });
 
-        const changedStory = Story.editStory(story, newAttributes);
+        describe(`when story status is ${UNSTARTED}`, () => {
+          describe(`when new story type is feature`, () => {
```

Comment:
You can apply this style https://github.com/Codeminer42/cm42-central/pull/576/files#diff-e7cad8ffcafac32387e195a4957fe897R257-R273 here.